qiskit/pulse/schedule.py

```diff
@@ -186,6 +188,87 @@ def flatten(self) -> 'ScheduleComponent':
         """Return a new schedule which is the flattened schedule contained all `instructions`."""
         return ops.flatten(self)
 
+    def filter(self, *filter_funcs: List[Callable],
+               channels: Iterable[int] = [],
+               instruction_types: Iterable[abc.ABCMeta] = [],
```

Comment:
Not sure about this type, tbh. It's not an `Instruction` instance, it's an `Instruction` subclass (used as a type). This is the result of, for example, `type(PulseInstruction)`.

Reply:
I don't think that's right, because:
`isinstance(PulseInstruction, Instruction)` is `False`. An instance of a `PulseInstruction` is an `Instruction`, but the type `PulseInstruction` is not an `Instruction`. I guess we can go with the output from `type`

Reply:
Ahh, you're right. I now understand what you mean. I believe in this case the [typing `Type` exists](https://docs.python.org/3/library/typing.html#typing.Type).